### PR TITLE
update emplaceRef API

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: src/core/internal/lifetime.d [hunk 1]
```diff
@@ -10,69 +10,74 @@ build.
 Furthermore, emplaceRef optionally takes a type parameter, which specifies
 the type we want to build. This helps to build qualified objects on mutable
 buffer, without breaking the type system with unsafe casts.
+
+Note: The function is used in mir-core.
```
Comment (Geod24) [inline src/core/internal/lifetime.d]: How come ? It is in `internal`, and *explicitly* defined as "for druntime internal use"

Comment (9il) [reply src/core/internal/lifetime.d]: This is why the note is here. It is a really useful API function for ref-counted libraries. It is used for a long time, so I just removed mir-core copy of the DRuntime code and start use these API directly.

Comment (9il) [reply src/core/internal/lifetime.d]: This MR is the next step required to remove the mir-core definitions.

Comment (atilaneves) [reply src/core/internal/lifetime.d]: If it's a useful API for ref-counted libraries, shouldn't it be in Phobos instead? Or, better yet, code.dlang.org?

In any case, why would druntime have a comment about one of its functions being used in a 3rd party library?

Comment (9il) [reply src/core/internal/lifetime.d]: nothing has been changed in the community

Comment (atilaneves) [reply src/core/internal/lifetime.d]: > nothing has been changed in the community

I don't know what you mean by that.

**Pull request closed without merging**